cmd/capibmadm/cmd/powervs/network/network.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+)
+
+// NewNetworkCommand function to add PowerVS network commands.
+func NewNetworkCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "network",
+		Short: "Perform PowerVS network operations",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// we cannot have multiple PersistentPreRunE for a command
+			// This block is a workaround mentioned in https://github.com/spf13/cobra/issues/252
+			root := cmd
+			for ; root.HasParent(); root = root.Parent() {
+			}
```

Comment:
what will happen if there is `PersistentPreRunE` also present in the `powervs` command as well? wondering if we can call `PersistentPreRunE` in the for loop itself so that it calls every function in the chain! wdyt?

Reply:
Then we should define `PersistentPreRunE` in all commands right Otherwise if we try to call `PersistentPreRunE` on commands which does not define it will panic.

Reply:
also if we want to call explicitly any subcommands persistenPrerun we can like this
```
			for ; root.HasParent(); root = root.Parent() {
				if root.Name() == "powervs" {
					if err := root.PersistentPreRunE(cmd, args); err != nil {
						return err
					}
					break
				}
			}
```

Reply:
What if we will call only if it is defined?
>Then we should define PersistentPreRunE in all commands right Otherwise if we try to call PersistentPreRunE on commands which does not define it will panic.

Reply:
We can do that, Should we implement it now or can we take it when its required.

Reply:
wondering if we can have that now so that people can just start using that code wherever required.

Reply:
I find following ways of doing it

From leaf command without need of ordering
```
PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
			root := cmd.Parent()
			for ; root.HasParent(); root = root.Parent() {
				if err := root.PersistentPreRunE(cmd, args); err != nil {
					return err
				}
			}
			if err := root.PersistentPreRunE(cmd, args); err != nil {
				return err
			}
			}
//output
network prerun
powervs prerun
capibmadm prerun
create prerun
```

From leaf command with of ordering
```
PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
			var preRunOrder []*cobra.Command
			for ; root.HasParent(); root = root.Parent() {
				preRunOrder = append(preRunOrder, root)
			}
			preRunOrder = append(preRunOrder, root)
			
			for i := len(preRunOrder) - 1; i >= 0; i-- {
				if err := preRunOrder[i].PersistentPreRunE(cmd, args); err != nil {
					return err
				}
			}
			}
//output
capibmadm prerun
powervs prerun
network prerun
create prerun
```

from any intermediate command
```
PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
                       root := cmd
			for ; root.HasParent(); root = root.Parent() {
				if root.PersistentPreRunE == nil || root.Name() == "network" {
					continue
				}
				if err := root.PersistentPreRunE(cmd, args); err != nil {
					return err
				}
			}
			// this is to run capibm command's prerun
			if err := root.PersistentPreRunE(cmd, args); err != nil {
				return err
			}
			}
//output
powervs prerun
capibmadm prerun
```

Reply:
first one looks good to me

Reply:
Updated it accordingly

Reply:
@mkumatag since we are just checking the service instance id is passed or not, can we make use of 	cmd.MarkFlagRequired() instead of persistantprerun?

Reply:
but you might need to use `MarkPersistentFlagRequired`, check which one suites the requirement.

Reply:
yeah, We are supposed to use `MarkPersistentFlagRequired` only.

Reply:
Updated it accrodingly